### ISSUE TITLE
Fine-tuned performance gap in Libero long

### ISSUE THREAD

**Issue by WalkerRusher** [baaivision/UniVLA#15]
Thanks for your great work!

I followed your listed steps to finetune the model  in Libero long.

However, there is clear performance gap between my reproduced results and the paper.

According to the paper, the success rate in libero long is 94.0%.

However, my result only achieved 87.4%. 

<img width="1274" height="524" alt="Image" src="https://github.com/user-attachments/assets/bbdeb5d7-325a-402c-a696-db76abbba088" />

Any suggestions would be deeply appreciated.

**Comment by Robertwyq** [comment]
@WalkerRusher I fine-tuned the model jointly on all four tasks. Did you only fine-tune on LIBERO-Long?

**Comment by WalkerRusher** [comment]
”does the provided checkpoint yield the expected results on your side?“

I am a little bit confused about "expected results" here. If you mean the success rate in libero long, I observe near 7 percent performance gap between the paper reported result and my reproduced result.

I believe I followed the finetune steps described in this repo (I started to finetune the model by using the provided world model parameters in all four libero tasks), yet I have no idea what cause the performance gap.

**Comment by Robertwyq** [comment]
@WalkerRusher By the way, did you modify here for video mode

<img width="1076" height="580" alt="Image" src="https://github.com/user-attachments/assets/869f92f9-c112-43b2-81a5-36c586a11a11" />

**Comment by fx-hit** [comment]
> [@WalkerRusher](https://github.com/WalkerRusher) Sorry for the confusion—what I meant was: can the provided checkpoint below reproduce the expected scores on your end? This would help me determine whether the issue lies with the simulation environment.
> 
> <img alt="Image" width="1470" height="342" src="https://private-user-images.githubusercontent.com/39329846/471462011-0d5fe29d-7b40-48ba-91dc-d132745c7fba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ1Njg1MTksIm5iZiI6MTc1NDU2ODIxOSwicGF0aCI6Ii8zOTMyOTg0Ni80NzE0NjIwMTEtMGQ1ZmUyOWQtN2I0MC00OGJhLTkxZGMtZDEzMjc0NWM3ZmJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA3VDEyMDMzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVlNTc0N2Y4NzlmYTY1ZTk0NmUxYWMyYzdmYjA0YTVjYWMzNTJhYjRjNzUzMGMzNTEyNTM0MDAxNDAxYzU0NjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ngpPL39sne3Xz5NDRP4YXDy2jTFR_p0l-rAdKdgG_NI">

I used the UNIVLA_LIBERO_VIDEO_BS192_8K checkpoint and the default random seed setting seed_everything(0, workers=True) in reference/RoboVLMs/eval/libero/evaluate_libero_emu.py. The evaluation results were:

Spatial: 0.9500
Object: 0.9580
Goal: 0.9220
Long: 0.8980
Average: 0.9320

For Long, only the tasks from libero_10 were tested. I couldn't reproduce the results reported in the paper or the table in the GitHub README.

**Comment by Robertwyq** [comment]
@fx-hit did you modify the video_mode to True?

**Comment by fx-hit** [comment]
> [@fx-hit](https://github.com/fx-hit) did you modify the video_mode to True?

Indeed not, I will later modify self.video_mode=True to test it. 
Additionally, after fine-tuning the world model-based checkpoint on the bridgev2 dataset, we found that the performance even surpassed the data reported in the paper, achieving an average success rate of 0.7604 in the simplerenv-windowx simulation.

**Comment by Robertwyq** [comment]
@fx-hit 
Using a model trained on video data but running image-only inference can indeed cause some performance degradation for long-horizon tasks.
Interesting — could you share the detailed results for each task so we can analyze the differences more thoroughly?

**Comment by fx-hit** [comment]
> [@fx-hit](https://github.com/fx-hit) Using a model trained on video data but running image-only inference can indeed cause some performance degradation for long-horizon tasks. Interesting — could you share the detailed results for each task so we can analyze the differences more thoroughly?

Put Spoon | Put Carrot | Stack Block | Put Eggplant | AVG
0.8333 | 0.7083 | 0.5833 | 0.9167 | 0.7604

**Comment by Robertwyq** [comment]
@fx-hit It seems that most of the improvement comes from the stack task. We found that the performance of this task tends to be somewhat unstable.
